sound/soc/sof/intel/mtl.c

```diff
@@ -672,13 +673,22 @@ static int mtl_dsp_core_put(struct snd_sof_dev *sdev, int core)
 struct snd_sof_dsp_ops sof_mtl_ops;
 EXPORT_SYMBOL_NS(sof_mtl_ops, SND_SOC_SOF_INTEL_HDA_COMMON);
 
+static int mtl_hda_dsp_probe_early(struct snd_sof_dev *sdev)
+{
+	snd_hdac_i915_bind(sof_to_bus(sdev), 0);
+	return hda_dsp_probe_early(sdev);
```

Comment:
`sof_to_bus()` referred to `sdev->pdata->hw_pdata`,  but `hw_pdata` is allocated and assigned in `hda_dsp_probe_early()`, is this safe?

Reply:
You gave me a good scare @aiChaoSONG, but it's fine really: the sof_to_bus(dev) argument is not used.

````c
void snd_hdac_i915_bind(struct hdac_bus *bus, int i915_bind)
{
	gpu_bind = i915_bind;
}
````

so we could remove this argument to make sure it's never used. Care to send a patch?